packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx

```diff
@@ -27,34 +26,37 @@ import {
   useConditionsSearch,
 } from './conditions.resource';
 import styles from './conditions-form.scss';
-
+import { useFormContext, Controller } from 'react-hook-form';
+import { ConditionFormData } from './conditions-form.component';
 interface ConditionsWidgetProps {
   closeWorkspace?: () => void;
   conditionToEdit?: ConditionDataTableRow;
-  formContext?: 'creating' | 'editing';
+  editing?: boolean;
   patientUuid: string;
   setHasSubmissibleValue?: (value: boolean) => void;
   setErrorCreating?: (error: Error) => void;
   setErrorUpdating?: (error: Error) => void;
-  submissionNotifier: BehaviorSubject<{ isSubmitting: boolean }>;
+  isSubmittingForm: boolean;
+  setIsSubmittingForm: Dispatch<boolean>;
 }
 
 const ConditionsWidget: React.FC<ConditionsWidgetProps> = ({
   closeWorkspace,
   conditionToEdit,
-  formContext,
+  editing,
   patientUuid,
-  setHasSubmissibleValue,
-  setErrorCreating: setErrorCreating,
-  setErrorUpdating: setErrorUpdating,
-  submissionNotifier,
+  isSubmittingForm,
+  setIsSubmittingForm,
+  setErrorCreating,
+  setErrorUpdating,
 }) => {
   const { t } = useTranslation();
   const { conditions, mutate } = useConditions(patientUuid);
-  const editing = formContext === 'editing';
+  const { control, watch, getValues, formState } = useFormContext<ConditionFormData>();
```

Comment:
using the formcontext and methods   [api ](https://react-hook-form.com/docs/useformcontext)